ingester/database.py

```diff
@@ -122,6 +255,32 @@ def insert_xai_analysis_metadata(self, model_card_id, xai_id, xai_analysis_metad
                     """
             session.run(query, xai_id=xai_id, mc_id=model_card_id)
 
+    def update_xai_analysis_metadata(self, model_card_id, xai_id, xai_analysis_metadata):
+        xai_name = model_card_id + "xai_analysis"
+
+        with self.driver.session() as session:
+            # Match the existing ExplainabilityAnalysis node
+            query = """
+            MATCH (xai:ExplainabilityAnalysis {external_id: $id})
+            SET xai.name = $name
+            """
+            session.run(query, name=xai_name, id=xai_id)
+
+            # Update existing node properties
+            for key, value in xai_analysis_metadata.items():
+                key = key.replace(" ", "_")
+                query = f"""
+                            MATCH (xai:ExplainabilityAnalysis {{external_id: $bias_id}})
+                            SET xai.{key} = $value
+                            """
+                session.run(query, bias_id=xai_id, value=value)
+
+            # query = """
+            #         MATCH (xai:ExplainabilityAnalysis {external_id: $xai_id}), (mc:ModelCard {external_id: $mc_id})
+            #         CREATE (xai)<-[:XAI_ANALYSIS]-(mc)
+            #         """
+            # session.run(query, xai_id=xai_id, mc_id=model_card_id)
+
     def connect_datasheet_mc(self, datasheet_id, mc_id):
```

Comment:
if you are updating xai and bias, you should also add the update for requirements.